File: Utils/Builder/src/bc/assets/types/Animation.java
package bc.assets.types;

import java.util.ArrayList;
import java.util.List;

import bc.assets.Asset;

public class Animation extends Asset
{
	private String name;
	private Texture texture;
	private List<AnimationFrame> frames;

	public Animation(String name, Texture texture)
	{
		this.name = name;
		this.texture = texture;
		frames = new ArrayList<AnimationFrame>();
	}
	
	public String getName()
	{
		return name;
	}
	
	public Texture getTexture()
	{
		return texture;
	}
	
	public void addFrame(AnimationFrame frame)
	{
		frames.add(frame);
	}
	
	public List<AnimationFrame> getFrames()
	{
		return frames;
	}
	
	public static class AnimationFrame
	{
		public int x;
		public int y;
		public int ox;
		public int oy;
		public int w;
		public int h;
	}
}

File: Bomberman/Bomberman/Content/Animation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BomberEngine.Core.Assets;
using BomberEngine.Core.Assets.Types;
using BomberEngine.Debugging;
using BomberEngine.Core.Visual;
using Microsoft.Xna.Framework;

namespace Bomberman.Content
{
    public class Animation : Asset
    {
        public String name;
        public TextureImage texture;

        public AnimationFrame[] frames;

        public Animation(String name)
        {
            this.name = name;
        }

        public void Draw(Context context, int frameIndex, float x, float y, Color color)
        {
            Rectangle src;
            int ox = frames[frameIndex].ox;
            int oy = frames[frameIndex].oy;

            src.X = frames[frameIndex].x;
            src.Y = frames[frameIndex].y;
            src.Width = frames[frameIndex].w;
            src.Height = frames[frameIndex].h;

            context.DrawImagePart(texture, src, x - ox, y - oy, color);
        }

        protected override void OnDispose()
        {
            if (!texture.IsDisposed())
            {
                texture.Dispose();
            }
        }
    }

    public struct AnimationFrame
    {
        public int x;
        public int y;
        public int ox;
        public int oy;
        public int w;
        public int h;
        public float duration;
    }
}
